app/controllers/collection_show_controller.rb

```diff
@@ -81,4 +84,18 @@ def collection
     @collection ||= Collection.find_by_friendlier_id!(params[:collection_id])
   end
   helper_method :collection
+
+
+  # Suppress certain Blacklight errors in the
+  # #facet action; respond instead with :unprocessable_entity.
+  def handle_unpermitted_params
+    # This only applies to #facet.
+    raise if params['action'] != 'facet'
+    # If the facet page param is a string (this is the normal case)
+    # but we are somehow *still* getting an UnpermittedParameters, reraise.
+    raise if params["facet.page"].is_a? String
+    # Suppress if the param is e.g. an array or hash.
+    return render plain: "Error: unpermitted parameters.", status: :unprocessable_entity
```

Comment:
Aha, I think I understand what's going on now, thanks, and sorry for misreading it before! `raise if params['action'] != 'facet'` is good. 

Would it be better/safer to `raise unless params["facet.page"].is_a? Array`?  I'm not sure what's in the middle of the two versions, or if it matters, but checking for what we actually mean may avoid having to think about it?

(I also could still go along with just swallowing _all_ of the unpermitted params honestly, but this is prob safer)

Reply:
No problem - this is super confusing. Here's what I see this code doing:

We know that `facet.page` is supposed to contain a string (it's just meant to be the page number as a string).

So the intent in this handler is to suppress Blacklight's `UnpermittedParameters` errors for arrays, hashes, anything passed into that param that *isn't* a string. That should cover the various URLs caught in the HB error- all of them are variations that involve passing non-strings to that parameter.

By the time we get to the `handle_unpermitted_params` handler, we already know we have bad params, of course.

Now, in this context, if `facet.page` *is* a regular string, and we're **still** somehow getting `UnpermittedParameters`  error, I **do** want to reraise, as this isn't the problem we're targeting.